### PR TITLE
Enable E2E tests for msbuild based projects

### PR DESCRIPTION
**Pull request by prafullbhosale**
cc @natemcmaster @NTaylorMullen

### CODE REVIEW

File: test/Shared/MsBuildProjectStrings.cs [hunk 1]
```diff
@@ -78,7 +79,7 @@ internal class MsBuildProjectStrings
       <Version>1.0.0-*</Version>
     </PackageReference>
     <PackageReference Include=""Microsoft.NET.Sdk"">
-      <Version>1.0.0-*</Version>
+      <Version>1.0.0-alpha-20161029-1</Version>
```
Comment (NTaylorMullen) [inline test/Shared/MsBuildProjectStrings.cs]: `*` not work?

File: test/E2E_Test/E2ETestBase.cs [hunk 2]
```diff
@@ -12,24 +13,26 @@ namespace Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test
 {
     public class E2ETestBase
```
Comment (NTaylorMullen) [inline test/E2E_Test/E2ETestBase.cs]: Lot of extra changes in this PR. Is this the bulk of what was changed to enable E2E tests?

File: test/E2E_Test/E2ETestBase.cs [hunk 3]
```diff
@@ -12,24 +13,26 @@ namespace Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test
...
     {
-        public static string BaseLineFilesDirectory = Path.Combine(Directory.GetCurrentDirectory(), "Baseline");
-
+        private static object syncObject = new object();
+        public string BaseLineFilesDirectory;
         public const string E2ESkipReason = "Disabling E2E test";
         public  const string codegeneratorToolName = "aspnet-codegenerator";
 
+        //protected string BaseLineFilesDirectory { get; }
```
Comment (NTaylorMullen) [inline test/E2E_Test/E2ETestBase.cs]: nuke

File: test/E2E_Test/E2ETestBase.cs [hunk 4]
```diff
@@ -12,24 +13,26 @@ namespace Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test
 {
     public class E2ETestBase
     {
-        public static string BaseLineFilesDirectory = Path.Combine(Directory.GetCurrentDirectory(), "Baseline");
-
+        private static object syncObject = new object();
```
Comment (NTaylorMullen) [inline test/E2E_Test/E2ETestBase.cs]: Don't use this.

File: test/E2E_Test/E2ETestBase.cs [hunk 5]
```diff
@@ -12,24 +13,26 @@ namespace Microsoft.VisualStudio.Web.CodeGeneration.E2E_Test
...
         public  const string codegeneratorToolName = "aspnet-codegenerator";
 
+        //protected string BaseLineFilesDirectory { get; }
         protected string TestProjectPath { get; set; }
         protected ITestOutputHelper Output { get; set; }
-        
 
         public E2ETestBase(ITestOutputHelper output)
         {
+            BaseLineFilesDirectory = Path.Combine(Directory.GetCurrentDirectory(), "Baseline");
```
Comment (NTaylorMullen) [inline test/E2E_Test/E2ETestBase.cs]: Why'd this have to change? You modify the working directory before calling into this now?

File: test/Shared/MsBuildProjectStrings.cs (hunk 1 above)
Comment (prafullbhosale) [inline test/Shared/MsBuildProjectStrings.cs]: The `3892` cli doesn't seem to be able to restore the newer version of sdk. 
😞

File: test/E2E_Test/E2ETestBase.cs (hunk 5 above)
Comment (prafullbhosale) [inline test/E2E_Test/E2ETestBase.cs]: Earlier was using the BaseLineFilesDirectory for retrieving content files. 
That's making the tests flaky. So have given up that approach and changed to using the baseline files as resources of the test assembly. 

Will remove this.

File: test/E2E_Test/E2ETestBase.cs (hunk 2 above)
Comment (prafullbhosale) [inline test/E2E_Test/E2ETestBase.cs]: Yes, this is the bulk of it. 
The renames of txt files are just to use them as resources of the assembly.

**Comment by prafullbhosale** [comment]
:up: :date:

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**